variables.tf

```
@@ -1,6 +1,163 @@
 variable "bucket" {
   description = "The name of the bucket"
   type        = string
+  default     = null
+  validation {
+    condition     = var.bucket != null ? length(var.bucket) <= 63 : true
+    error_message = "Bucket name must be <= to 63 character in length."
+  }
+}
+
+variable "logging" {
+  description = "Schema object for the S3 bucket logging configuration"
+  type = object({
+    target_bucket         = string # (Required) The name of the bucket where you want Amazon S3 to store server access logs.
+    target_prefix         = string # (Required) A prefix for all log object keys.
+    expected_bucket_owner = string # (Optional, Forces new resource) The account ID of the expected bucket owner.
+    target_grants = list(object({
+      grantee = object({
+        email_address = string # (Optional) Email address of the grantee. See Regions and Endpoints for supported AWS regions where this argument can be specified.
+        id            = string # (Optional) The canonical user ID of the grantee.
+        type          = string # (Required) Type of grantee. Valid values: CanonicalUser, AmazonCustomerByEmail, Group.
+        uri           = string # (Optional) URI of the grantee group.
+      })
+      permission = string # (Required) Logging permissions assigned to the grantee for the bucket. Valid values: FULL_CONTROL, READ, WRITE.
+    }))
+  })
+  default = null
+}
+
+variable "ownership_controls" {
+  description = "List of schema objects for the S3 ownership controls"
+  type = object({
+    rule = object({             # (Required) Configuration block with Ownership Controls rules.
+      object_ownership = string # (Required) Object ownership. Valid values: BucketOwnerPreferred, ObjectWriter or BucketOwnerEnforced
+    })
+  })
+  default = null
+}
+
+variable "request_payment_configuration" {
+  description = "Request payment configuration for the S3 bucket"
+  type = object({
+    expected_bucket_owner = string # (Optional, Forces new resource) The account ID of the expected bucket owner.
+    payer                 = string # (Required) Specifies who pays for the download and request fees. Valid values: BucketOwner, Requester.
+  })
+  default = null
+}
+
+variable "cors_configuration" {
+  description = "List of cors rules the S3 bucket"
```

"List" implies the resource type is a list, but this is an object... Suggest:

```suggestion
  description = "Schema object of CORS configurations for the S3 bucket"
```